File: android/src/main/java/smoksmog/logger/AggregatingLogger.java
package smoksmog.logger;


import java.util.Arrays;
import java.util.Collection;

public class AggregatingLogger implements Logger {

    private final Collection<Logger> loggerCollection;

    public AggregatingLogger(Logger... logger) {
        this(Arrays.asList(logger));
    }

    public AggregatingLogger(Collection<Logger> loggerCollection) {
        this.loggerCollection = loggerCollection;
    }

    @Override
    public void v(String tag, String message) {
        for (Logger logger : loggerCollection) {
            logger.v(tag, message);
        }
    }

    @Override
    public void v(String tag, String message, Throwable throwable) {
        for (Logger logger : loggerCollection) {
            logger.v(tag, message, throwable);
        }
    }

    @Override
    public void d(String tag, String message) {
        for (Logger logger : loggerCollection) {
            logger.d(tag, message);
        }
    }

    @Override
    public void d(String tag, String message, Throwable throwable) {
        for (Logger logger : loggerCollection) {
            logger.d(tag, message, throwable);
        }
    }

    @Override
    public void i(String tag, String message) {
        for (Logger logger : loggerCollection) {
            logger.i(tag, message);
        }
    }

    @Override
    public void i(String tag, String message, Throwable throwable) {
        for (Logger logger : loggerCollection) {
            logger.i(tag, message, throwable);
        }
    }

    @Override
    public void w(String tag, String message) {
        for (Logger logger : loggerCollection) {
            logger.w(tag, message);
        }
    }

    @Override
    public void w(String tag, String message, Throwable throwable) {
        for (Logger logger : loggerCollection) {
            logger.w(tag, message, throwable);
        }
    }

    @Override
    public void e(String tag, String message) {
        for (Logger logger : loggerCollection) {
            logger.e(tag, message);
        }
    }

    @Override
    public void e(String tag, String message, Throwable throwable) {
        for (Logger logger : loggerCollection) {
            logger.e(tag, message, throwable);
        }
    }
}

File: android/src/main/kotlin/smoksmog/logger/AggregatingLogger.kt
package smoksmog.logger


import java.util.Arrays

class AggregatingLogger(private val loggerCollection: Collection<Logger>) : Logger {

    constructor(vararg logger: Logger) : this(Arrays.asList(*logger)) {
    }

    override fun v(tag: String, message: String) {
        for (logger in loggerCollection) {
            logger.v(tag, message)
        }
    }

    override fun v(tag: String, message: String, throwable: Throwable) {
        for (logger in loggerCollection) {
            logger.v(tag, message, throwable)
        }
    }

    override fun d(tag: String, message: String) {
        for (logger in loggerCollection) {
            logger.d(tag, message)
        }
    }

    override fun d(tag: String, message: String, throwable: Throwable) {
        for (logger in loggerCollection) {
            logger.d(tag, message, throwable)
        }
    }

    override fun i(tag: String, message: String) {
        for (logger in loggerCollection) {
            logger.i(tag, message)
        }
    }

    override fun i(tag: String, message: String, throwable: Throwable) {
        for (logger in loggerCollection) {
            logger.i(tag, message, throwable)
        }
    }

    override fun w(tag: String, message: String) {
        for (logger in loggerCollection) {
            logger.w(tag, message)
        }
    }

    override fun w(tag: String, message: String, throwable: Throwable) {
        for (logger in loggerCollection) {
            logger.w(tag, message, throwable)
        }
    }

    override fun e(tag: String, message: String) {
        for (logger in loggerCollection) {
            logger.e(tag, message)
        }
    }

    override fun e(tag: String, message: String, throwable: Throwable) {
        for (logger in loggerCollection) {
            logger.e(tag, message, throwable)
        }
    }
}
